docs/lang/articles/basic/field.md

```diff
@@ -3,100 +3,157 @@ sidebar_position: 3
 
-Fields are **global** variables provided by Taichi. **Global** indicates that fields can be read/written from both the Python scope and the Taichi scope. A field can be considered as a multi-dimensional array of elements, and it can be either **dense** or **sparse**. Similar to a NumPy `ndarray` object, a field has a data type and a shape. Moreover, an element of a field can be a scalar, a **vector**, a **matrix**, or a **struct**.
+A field has its own data type and shape and can be considered as a multi-dimensional array of elements.
+An element of a field can be a **scalar**, a **vector**, a **matrix**, or a **struct**.
+The sparsity of a field element is **dense** by default, but it can also be **sparse**, as detailed described in [Sparse computation](/lang/articles/advanced/sparse).
 
-The term **field** is borrowed from mathematics and physics. If you
-have already known [scalar field](https://en.wikipedia.org/wiki/Scalar_field) (e.g., heat field) or vector field (e.g., [gravitational field](https://en.wikipedia.org/wiki/Gravitational_field)) in mathematics and physics, it will be straightforward to understand the fields in Taichi.
-
-To be noticed:
-* Fields are always accessed by indices.
-* Field values are initially zero.
-* Sparse fields are initially inactive.
-
-:::tip
-In earlier versions of Taichi, you could not allocate new fields after executing the first kernel. Since Taichi v0.8.0, you can use a new class `FieldsBuilder` for dynamic field allocation and destruction. For more details, please see [Field (advanced)](/lang/articles/advanced/layout).
+:::note
+The term **field** is borrowed from mathematics and physics.
+If you have already known [scalar field](https://en.wikipedia.org/wiki/Scalar_field) (e.g., heat field) or vector field (e.g., [gravitational field](https://en.wikipedia.org/wiki/Gravitational_field)) in mathematics and physics,
+it will be straightforward to understand the fields in Taichi.
 :::
 
 ## Scalar fields
+We start introducing fields from this very basic type, the elements of scalar fields are simply scalars.
+* A 0D scalar field is a single scalar.
+* A 1D scalar field is an array.
+* A 2D scalar field can be used to represent a 2D regular grid of values.
+* A 3D scalar field can be used for volumetric data.
 
-A simple example might help you understand scalar fields. Assume you have a rectangular wok on the top of a fire. At each point of the wok, there would be a temperature. The surface of the wok forms a heat field. The width and height of the wok are similar to the `shape` of the Taichi scalar field. The temperature (0-D scalar) is like the element of the Taichi scalar field. We could use the following field to represent the
-heat field on the wok:
-
+### Declaration
 ``` python
-heat_field = ti.field(dtype=ti.f32, shape=(width_wok, height_wok))
+import taichi as ti
+ti.init(arch=ti.cpu)
+
+gravity          = ti.field(ti.f32, shape=())           # 0-D
```

Comment:
Gravity is a 3D vector. Energy is more suitable to be represented as a scalar field.

`energy = ti.field(ti.f32, shape=())`